src/bsp/src/stm32/include/Decawave.h

```diff
@@ -0,0 +1,163 @@
+#ifndef __DECAWAVE_H__
+#define __DECAWAVE_H__
+
+#include "UWBRxFrame.h"
+#include "deca_device_api.h"
+#include "deca_port.h"
+#include <BaseTask.h>
+#include <FreeRTOS.h>
+#include <functional>
+#include <memory>
+#include <task.h>
+
+enum class DW_LED { LED_0, LED_1, LED_2, LED_3 };
+enum class UWBSpeed { SPEED_110K, SPEED_850K, SPEED_6M8 };
+
+class Decawave {
+  public:
+    explicit Decawave(decaDevice_t spiDevice);
```

Comment:
Why the explicit? Also why two constructor?

Reply:
The explicit comes from a clang suggestion that single argument constructors should be marked explicit.
As for the second constructor, when I wrote the class I wasn't sure (and still am not as to how we would construct these objects ie configure them at runtime or through the constructor) so I left both options. It's probably going to end up being the second constructor with our default values and sometimes change them at runtime for tests

Reply:
I personally prefer at the construction time, but it's fine keep both too